File: Medium/1838. Frequency of the Most Frequent Element/1838. Frequency of the Most Frequent Element.ts
function maxFrequency(nums: number[], k: number): number {
    nums.sort((a, b) => a - b); // Sort the array

    let maxFreq = 0;
    let left = 0;
    let sum = 0;

    for (let right = 0; right < nums.length; right++) {
        sum += nums[right];

        // Check if we need to shrink the window
        while ((right - left + 1) * nums[right] - sum > k) {
            sum -= nums[left];
            left += 1;
        }

        // Update the maximum frequency
        maxFreq = Math.max(maxFreq, right - left + 1);
    }

    return maxFreq;
}

File: Medium/1838. Frequency of the Most Frequent Element/1838. Frequency of the Most Frequent Element.py
from typing import List

class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()  # Sort the array

        max_freq = 0
        left = 0
        sum_val = 0

        for right in range(len(nums)):
            sum_val += nums[right]

            # Check if we need to shrink the window
            while (right - left + 1) * nums[right] - sum_val > k:
                sum_val -= nums[left]
                left += 1

            # Update the maximum frequency
            max_freq = max(max_freq, right - left + 1)

        return max_freq
